vinyldns/vinyldns-python
# Add Unit Tests Support

**To test this**
1. checkout this PR
1. `rm -rf .virtualenv` in the event you already have one
1. `./bootstrap.sh` which will reset your virtualenv
1. `tox -e py36` to run everything, should be green

**Summary**
This is a lean version of a full build.

* Added unit tests using responses, a simple way to mock HTTP api calls.  Here is the responses library, very active - https://github.com/getsentry/responses
* Added a setup.cfg to store all "configurations" instead of having separate files
* Added PyTest command to setup.py to simplify running unit tests
* Added tox support, which is a multi-environment manager for different versions
* Added tox support to the travis build
* Added code coverage support

**Note: I am not publishing to codecov yet, that can be a simple follow on.**
